src/plugins/analysis/file_system_metadata/code/file_system_metadata.py

```diff
@@ -98,31 +75,45 @@ def _parent_has_file_system_metadata(self, file_object: FileObject) -> bool:
     def _extract_metadata(self, file_object: FileObject):
         file_type = file_object.processed_analysis['file_type']['mime']
         if file_type in self.FS_MIME_TYPES:
-            self._extract_metadata_from_file_system(file_object, file_type)
+            self._extract_metadata_from_file_system(file_object)
         elif file_type in self.ARCHIVE_MIME_TYPES:
             self._extract_metadata_from_tar(file_object)
         if self.result:
-            file_object.processed_analysis['file_system_metadata'] = {'files': self.result}
+            file_object.processed_analysis[self.NAME] = {'files': self.result}
             self._add_tag(file_object, self.result)
 
-    def _extract_metadata_from_file_system(self, file_object: FileObject, file_type: str):
-        type_parameter = '-t {}'.format(file_type.split('/')[1])
-        with suppress(MountingError):
-            with mount(file_object.file_path, type_parameter, get_temp_dir_path(self.config)) as mounted_path:
-                self._analyze_metadata_of_mounted_dir(mounted_path)
+    def _extract_metadata_from_file_system(self, file_object: FileObject):
+        with TemporaryDirectory() as tmp_dir:
+            input_file = Path(tmp_dir) / "input.img"
+            input_file.write_bytes(file_object.binary or Path(file_object.file_path).read_bytes())
+            output = self._mount_in_docker(tmp_dir)
+            output_file = Path(tmp_dir) / "output.pickle"
+            if output_file.is_file():
+                self._analyze_metadata_of_mounted_dir(pickle.loads(output_file.read_bytes()))
+            else:
+                message = 'mount failed:\n{}'.format(output)
+                logging.warning('[{}] {}'.format(self.NAME, message))
+                file_object.processed_analysis[self.NAME]['failed'] = message
+
+    def _mount_in_docker(self, input_dir: str) -> str:
+        return run_docker_container(
+            DOCKER_IMAGE,
+            mount=('/work', input_dir),
+            label=self.NAME,
+            timeout=int(self.timeout * .8),
+            privileged=True
+        )
 
-    def _analyze_metadata_of_mounted_dir(self, mounted_dir: Path):
-        for file_ in safe_rglob(mounted_dir, False, False):  # FIXME files with PermissionError could be ignored
-            if file_.is_file() and not file_.is_symlink():
-                self._enter_results_for_mounted_file(file_)
+    def _analyze_metadata_of_mounted_dir(self, docker_results: Tuple[str, str, os.stat_result]):
+        for file_name, file_path, file_stats in docker_results:
+            self._enter_results_for_mounted_file(file_name, file_path, file_stats)
 
-    def _enter_results_for_mounted_file(self, file_: Path):
-        result = self.result[b64encode(file_.name.encode()).decode()] = {}
-        stats = os.lstat(str(file_))
+    def _enter_results_for_mounted_file(self, file_name: str, file_path: str, stats: os.stat_result):
+        result = self.result[b64encode(file_name.encode()).decode()] = {}
```

Comment:
What is the historical explanation behind this usage of base64?

Reply:
there was a problem with certain (special) characters in file names but I can't remember exactly what is was